src/SerialPort/SerialPort.ts

```diff
@@ -13,22 +17,42 @@ export const SERIALPORT_CHANNEL = {
     OPEN: 'serialport:open',
     CLOSE: 'serialport:close',
     WRITE: 'serialport:write',
+    UPDATE: 'serialport:update',
+    SET: 'serialport:set',
 
     ON_CLOSED: 'serialport:on-close',
     ON_DATA: 'serialport:on-data',
+    ON_UPDATE: 'serialport:on-update',
+    ON_SET: 'serialport:on-set',
+    ON_CHANGED: 'serialport:on-changed',
 
     IS_OPEN: 'serialport:is-open',
 };
 
 export const SerialPort = async (
     options: SerialPortOpenOptions<AutoDetectTypes>,
+    overwrite: boolean,
     onData: (data: Uint8Array) => void = () => {},
-    onClosed: () => void = () => {}
+    onClosed: () => void = () => {},
+    onUpdate: (newOptions: UpdateOptions) => void = () => {},
+    onSet: (newOptions: SetOptions) => void = () => {},
+    onChange: (
+        newOptions: SerialPortOpenOptions<AutoDetectTypes>
+    ) => void = () => {}
 ) => {
```

Comment:
With the addition of all these callbacks, I think we should bundle them into an object. This would:

1. Make it easier to understand what callback does what (instead of looking up “Ah, the third (or fourth?) callback is `onUpdate`”).
2. Make it easier to leave out callbacks you don't want to use.

So, my suggestion would turn https://github.com/NordicPlayground/pc-nrfconnect-serial-terminal/pull/1/files#diff-17d92a7c7f04d9e54c3d76a026e131f070d7fd773c035be10474e22e79fa0586R35-R50 from 

```ts
serialPort = await SerialPort(
  options as SerialPortOpenOptions<AutoDetectTypes>,
  overwrite,
  (data) => eventEmitter.emit('response', [data]),
  () => {},
  (newOptions) => dispatch(setSerialOptions(newOptions)),
  (newOptions) => dispatch(setSerialOptions(newOptions)),
  (newOptions) => {
    dispatch(setSerialOptions(newOptions));
    console.log(
      `Received new settings from serial port: ${JSON.stringify(newOptions)}`
    );
  }
);
```

into

```ts
serialPort = await SerialPort(
  options as SerialPortOpenOptions<AutoDetectTypes>,
  overwrite,
  {
    onData: (data) => eventEmitter.emit('response', [data]),
    onUpdate: (newOptions) => dispatch(setSerialOptions(newOptions)),
    onSet: (newOptions) => dispatch(setSerialOptions(newOptions)),
    onChange: (newOptions) => {
      dispatch(setSerialOptions(newOptions));
      console.log(
        `Received new settings from serial port: ${JSON.stringify(newOptions)}`
      );
    },
  }
);
```

To achieve this, I think you only have to change the signature of the function here to this:
```ts
export const SerialPort = async (
    options: SerialPortOpenOptions<AutoDetectTypes>,
    overwrite: boolean,
    {
        onData = () => {},
        onClosed = () => {},
        onUpdate = () => {},
        onSet = () => {},
        onChange = () => {},
    }: {
        onData?: (data: Uint8Array) => void;
        onClosed?: () => void;
        onUpdate?: (newOptions: UpdateOptions) => void;
        onSet?: (newOptions: SetOptions) => void;
        onChange?: (newOptions: SerialPortOpenOptions<AutoDetectTypes>) => void;
    } = {}
) => {
```

This looks more complex than the existing function signature, but nothing more needs to be changed in this function and I think it is worth it. :-)